csrc/fusion.cpp

```diff
@@ -331,6 +331,23 @@ void Fusion::replaceOutput(Val* output, Val* replacement) {
   // (https://github.com/csarofeen/pytorch/issues/1112)
   if (io_alias_.count(output) != 0) {
     auto input = io_alias_[output];
+    if (replacement->isFusionOutput()) {
```

Comment:
I'm not certain about having this check inside the if block.

If we just want to avoid changing semantics, we should check that `replacement` and old `output` should have the same alias info before the replacement. So I think it should be inside the previous if block where

`if (find_output != outputs_.end()) {`